pkg/metadata/provider_aws.go

```diff
@@ -61,6 +80,19 @@ func (p *ProviderAWS) Extract() ([]byte, error) {
 	return userData, nil
 }
 
+// lookup a value (lookupName) in aws metaservice and store in given fileName
+func awsMetaGet(lookupName string, fileName string, fileMode os.FileMode) {
+	lookupValue, err := awsGet(metaDataURL + lookupName)
```

Comment:
not sure if this is more idiomatic, but I've seen:
```
if lookupValue, err := awsGet(...); err == nil {
} else {
}
```
saves one line and `lookupValue` is slightly tighter scoped